core/platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/proxy/MachineExtensionProxyServlet.java

```diff
@@ -131,9 +133,18 @@ private String getExtensionApiUrl(HttpServletRequest req) throws NotFoundExcepti
         }
 
         final UriBuilder uriBuilder = UriBuilder.fromUri(server.getUrl())
-                                                .replacePath(req.getRequestURI())
                                                 .replaceQuery(req.getQueryString());
 
+        // here we cut workspaceId from extension API Url for getting access to org.eclipse.che.api.core.rest.ApiInfoService
+        // e.g localhost:8080/ide/ext/{wsid} -> localhost:{extServicesPort}/ide/ext/
```

Comment:
move this comment inside branch of if statement because it doesn't describe else branch